### PR TITLE
fix(issue): queued-shell phantom milestone promoted to active by deriveStateFromDb, no user-facing recovery path

### LINKED ISSUE #1524: queued-shell phantom milestone promoted to active by deriveStateFromDb, no user-facing recovery path
## Problem

A `queued` milestone with no content (no CONTEXT, ROADMAP, or SUMMARY files) and no slices — a phantom row left by `gsd_milestone_generate_id` that was never planned — gets promoted to "active" by the DB-backed state derivation. This makes `/gsd` show it as the current milestone, but the guided flow has nothing to offer (no directory, no files, no slices). The user cannot complete, park, or discard it through normal flows because:

1. The guided flow's "discard" option is buried in sub-menus that may not render correctly for a phantom.
2. There is no direct CLI command like `/gsd discard M015`.
3. The doctor explicitly skips `queued` milestones, so it never detects or flags the orphan.
4. `deleteMilestone()` exists in `milestone-actions.js` but is only accessible through the guided TUI or the `/gsd rethink` conversational flow.

### Bug 1: `state/derive/from-db.js:130-136` — queued-shell promoted to active without content check

```js
if (!activeMilestoneFound && firstDeferredQueuedShell) {
    const shell = firstDeferredQueuedShell;
    activeMilestone = { id: shell.id, title: shell.title };
    activeMilestoneSlices = [];
    activeMilestoneFound = true;
    if (shell.hasDraftContext)
        activeMilestoneHasDraft = true;
    const entry = registry.find(e => e.id === shell.id);
    if (entry)
        entry.status = 'active';
}
```

A `queued-shell` milestone (classified by `milestone-readiness.js:17`) is deferred during normal iteration, but then unconditionally promoted as a fallback when no other active milestone is found. A phantom (no content, no draft) should be skipped — only a queued-shell with draft context (i.e., `discuss-milestone` was started) should be promoted.

**Fix:** Add a content guard before promotion:
```js
if (!activeMilestoneFound && firstDeferredQueuedShell && firstDeferredQueuedShell.hasDraftContext) {
```

### Bug 2: `doctor-runtime-checks.js:732` — doctor skips queued milestones entirely

```js
if (milestone.status === "queued")
    continue;
```

The orphan DB-row check intentionally skips `queued` milestones, preventing the doctor from detecting phantoms that should be cleaned up.

**Fix:** Change to only skip `queued` milestones that have content files (i.e., legitimate in-flight planning):
```js
if (milestone.status === "queued") {
    const hasContent = resolveMilestoneFile(basePath, milestone.id, "CONTEXT")
        || resolveMilestoneFile(basePath, milestone.id, "ROADMAP")
        || resolveMilestoneFile(basePath, milestone.id, "SUMMARY");
    if (hasContent) continue;
    // Phantom queued row — report as orphan
}
```

## Expected Behavior

1. `deriveStateFromDb` should skip `queued-shell` milestones that have no content AND no draft context, treating them as ghosts rather than promoting them to active.
2. The doctor should detect `queued` milestones with no disk artifacts as orphaned/phantom rows and offer to clean them up (delete the DB row).
3. A direct `/gsd discard M015` CLI command would prevent this class of stuck state.

## Environment
- GSD version: 1.11.0
- Unit: M015 (queued, 0 slices, no content files)
- Forensics version: auto-generated

## Reproduction Context

1. `gsd_milestone_generate_id` is called (e.g., via `discuss-milestone` or planning), inserting a DB row with `status: "queued"`.
2. The milestone is never planned — no `gsd_plan_milestone` call follows.
3. All other milestones complete.
4. `deriveStateFromDb` finds no active milestone and promotes the phantom `queued-shell` to active.
5. User runs `/gsd` and gets stuck on an empty milestone with no actionable options.

## Forensic Evidence

- M015 DB row: `status: "queued"`, `sliceCount: 0`, no CONTEXT/ROADMAP/SUMMARY files, no milestone directory on disk.
- `isGhostMilestone()` in `state.js:45-64` correctly identifies this as a ghost, but `deriveStateFromDb()` in `state/derive/from-db.js` never calls it.
- `milestone-readiness.js:17` classifies as `queued-shell`.
- Doctor `doctor-runtime-checks.js:732` skips all `queued` milestones with `continue`.

---
*Auto-generated by `/gsd forensics`*

### PR DESCRIPTION
**Pull request by jeremymcs**
## Summary
- Guarded queued-shell promotion on draft context and made the doctor flag content-less queued phantoms as orphans; verified via derive-state, stale-queued, and doctor orphan test suites.

## Bugs Addressed
- [x] **`state/derive/from-db.js:130-136` — queued-shell promoted to active without content check**
- [x] **`doctor-runtime-checks.js:732` — doctor skips queued milestones entirely**

## Verification
- Completed in the repository worktree before push.

## Related Issue
- Closes #1524
- [#1524 queued-shell phantom milestone promoted to active by deriveStateFromDb, no user-facing recovery path](https://github.com/open-gsd/gsd-pi/issues/1524)

## User
- @TerminalSausage

## Repo
- `open-gsd/gsd-pi`

## Branch
- `issue/1524-queued-shell-phantom-milestone-promoted--1784690023`

---
<a href="https://app.blacksmith.sh/open-gsd/codesmith/gsd-pi/pr/1525"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is enabled.</sup>

---

> [!NOTE]
> **Medium Risk**
> Changes core DB-backed milestone selection and doctor orphan detection; behavior is well-covered by tests but affects how active milestone and next-action are derived for edge-case repos.
> 
> **Overview**
> Fixes **#1524**, where empty `queued` milestone rows from `gsd_milestone_generate_id` could become the active milestone and leave users with no clear recovery path.
> 
> **State derivation** (`deriveStateFromDb`) no longer promotes a content-less queued shell to active by default. Promotion is limited to shells that have **CONTEXT-DRAFT** or appear in **PROJECT.md**’s Milestone Sequence (legitimate roadmap stages). Phantom shells stay `pending`; when nothing is promotable, the UI uses **pre-planning** with guidance to run `/gsd doctor` instead of a misleading dependency **blocked** state.
> 
> **Doctor** applies `orphan_milestone_db` to **all** milestone statuses (including `queued`) when no milestone directory exists on disk, using new **`milestoneDirExists`** (directory presence) rather than **`resolveMilestonePath`** (content-bearing), so in-flight scaffold dirs are not false positives.
> 
> Tests cover phantoms, PROJECT.md promotion, doctor orphan cases, and masking of later draft-bearing shells.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 11fc160e0910202b6af1e11ac822c124d2542a90. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by jeremymcs** [comment]
🔎 **Verification — 0 of 2 addressed**

Re-checked PR #1525 against [#1524 queued-shell phantom milestone promoted to active by deriveStateFromDb, no user-facing recovery path](https://github.com/open-gsd/gsd-pi/issues/1524).

## Subtasks
- [ ] **`state/derive/from-db.js:130-136` — queued-shell promoted to active without content check** _(pending)_
- [ ] **`doctor-runtime-checks.js:732` — doctor skips queued milestones entirely** _(pending)_

_Re-run from Patchdeck to refresh._

File: src/resources/extensions/gsd/state/derive/from-db.ts [hunk 1]
```diff
@@ -214,12 +214,18 @@ async function buildRegistryAndFindActive(
...
+  // (i.e. discuss-milestone was started). A phantom shell with no content and
+  // no draft is a ghost row left by gsd_milestone_generate_id that was never
+  // planned — promoting it strands the user on an empty milestone with no
+  // actionable options (#1524). Leave it as 'pending' so state falls through
+  // to handleNoActiveMilestone and the doctor can flag it as an orphan.
+  if (!activeMilestoneFound && firstDeferredQueuedShell && firstDeferredQueuedShell.hasDraftContext) {
     const shell = firstDeferredQueuedShell;
     activeMilestone = { id: shell.id, title: shell.title };
     activeMilestoneSlices = [];
     activeMilestoneFound = true;
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: **Accepted** — queued for a code change.
**In progress** — applying the accepted fix.
**Verifying** — checking the applied changes.
**Resolved** — addressed in commit `aacc893`.

**Comment by jeremymcs** [comment]
Started an automated PR update.

Accepted review feedback is being applied.
Progress and final results will be posted in the related review thread(s).

File: src/resources/extensions/gsd/state/derive/from-db.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: Addressed in commit a29ad2de. The tracker no longer locks onto the first queued-shell unconditionally — it upgrades to the first shell that carries draft context:

```ts
if (
  !firstDeferredQueuedShell ||
  (!firstDeferredQueuedShell.hasDraftContext && readiness.hasDraftContext)
) {
  firstDeferredQueuedShell = { id: m.id, title, deps, hasDraftContext: readiness.hasDraftContext };
}
```

So an earlier phantom (no draft) no longer masks a later resumable draft milestone. Covered by the regression test `earlier phantom queued shell does not mask a later draft-bearing queued shell (#1524)` in `stale-queued-milestone.test.ts`, which asserts the later draft-bearing shell (M020) is promoted.

File: src/resources/extensions/gsd/state/derive/from-db.ts [hunk 2]
```diff
@@ -187,7 +187,16 @@ async function buildRegistryAndFindActive(
       }
 
       if (readiness.kind === 'queued-shell') {
-        if (!firstDeferredQueuedShell) {
+        // Track the deferred queued-shell used for the promotion guard below.
+        // Prefer the earliest shell WITH draft context: a phantom shell (no
+        // draft) must not mask a later resumable draft milestone (#1524).
+        // Store the first shell seen, but upgrade to a later shell once it
+        // carries draft context so a real draft isn't hidden behind an earlier
+        // phantom. Falls back to the earliest phantom when no draft exists.
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: Fixed in commit fbe77515. The comment was corrected to match the actual behavior — it no longer claims a phantom is promoted as a fallback:

```
// Store the first shell seen, then upgrade to the first shell that
// carries draft context. Only draft-bearing shells are eligible for
// promotion (see the guard below), so this upgrade ensures an earlier
// phantom (no draft) can't mask a later resumable draft milestone
// (#1524). A stored phantom is never promoted — it stays 'pending'.
```

File: src/resources/extensions/gsd/state/derive/from-db.ts [hunk 3]
```diff
@@ -214,12 +223,18 @@ async function buildRegistryAndFindActive(
...
   }
 
-  if (!activeMilestoneFound && firstDeferredQueuedShell) {
+  // Only promote a deferred queued-shell to active when it has draft context
+  // (i.e. discuss-milestone was started). A phantom shell with no content and
+  // no draft is a ghost row left by gsd_milestone_generate_id that was never
+  // planned — promoting it strands the user on an empty milestone with no
+  // actionable options (#1524). Leave it as 'pending' so state falls through
+  // to handleNoActiveMilestone and the doctor can flag it as an orphan.
+  if (!activeMilestoneFound && firstDeferredQueuedShell && firstDeferredQueuedShell.hasDraftContext) {
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: Good catch — addressed in commit aacc8938. `handleNoActiveMilestone` now distinguishes a genuine dependency block from a phantom-only repo. When no pending entry has unmet deps (i.e. the remaining rows are content-less queued shells), it no longer returns the misleading `blocked` / "resolve dependencies" state. Instead it returns `pre-planning` with a recovery-oriented nextAction pointing at the doctor:

```ts
if (blockerDetails.length > 0) {
  return buildDerivedState(context, 'blocked', 'Resolve milestone dependencies before proceeding.', { blockers: blockerDetails });
}
const phantomIds = pendingEntries.map(e => e.id).join(', ');
return buildDerivedState(
  context,
  'pre-planning',
  `Found queued milestone(s) with no planning content and no dependencies (${phantomIds}) — likely orphaned rows. Run /gsd doctor fix to repair them, or /gsd to plan a milestone.`,
);
```

This closes the "no user-facing recovery path" gap in #1524 — combined with the doctor change that now flags these rows as `orphan_milestone_db`. Covered by the new regression test `phantom-only repo surfaces a doctor recovery path, not a misleading dep-block message (#1524)`.

File: src/resources/extensions/gsd/state/derive/from-db.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: **Accepted** — queued for a code change.
**In progress** — applying the accepted fix.
**Verifying** — checking the applied changes.
**Resolved** — addressed in commit `aacc893`.

File: src/resources/extensions/gsd/state/derive/from-db.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: **Accepted** — queued for a code change.
**In progress** — applying the accepted fix.
**Verifying** — checking the applied changes.
**Resolved** — addressed in commit `aacc893`.

**Comment by jeremymcs** [comment]
Started an automated PR update.

Accepted review feedback is being applied.
Progress and final results will be posted in the related review thread(s).

File: src/resources/extensions/gsd/state/derive/from-db.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: Addressed in commit `aacc893`.

File: src/resources/extensions/gsd/state/derive/from-db.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: Addressed in commit `aacc893`.

Verified the queued-shell tracking comment at from-db.ts:189–195 was already corrected (commit fbe77515): it no longer claims a phantom is used as a fallback, and now accurately describes tracking the first shell, upgrading to the first draft-bearing shell, and that only draft-bearing shells are promoted while a stored phantom stays 'pending'. No further change needed.

File: src/resources/extensions/gsd/state/derive/from-db.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: Addressed in commit `aacc893`.

Verified the phantom-only recovery path was already added (commit aacc8938): handleNoActiveMilestone now returns 'blocked' only when a pending milestone has unmet deps, and otherwise returns a 'pre-planning' state directing the user to `/gsd doctor fix` or `/gsd` instead of the misleading dependency message. Covered by a passing regression test. No further change needed.

**Comment by jeremymcs** [comment]
## CI needs attention

Changes were pushed in commit `aacc893`, but CI/CD checks are still failing:

- **node22-smoke**: Check run cancelled ([details](https://github.com/open-gsd/gsd-pi/actions/runs/29889224068/job/88827211942))
- **windows-portability**: Check run cancelled ([details](https://github.com/open-gsd/gsd-pi/actions/runs/29889224068/job/88827211846))
- **ci-gate**: Check run failure ([details](https://github.com/open-gsd/gsd-pi/actions/runs/29889224068/job/88827211545))
- **build**: Check run cancelled ([details](https://github.com/open-gsd/gsd-pi/actions/runs/29889224068/job/88826232139))

Review the failing checks before merging.

File: src/resources/extensions/gsd/doctor-runtime-checks.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { basename, dirname, join } from "node:path";
 
 import type { DoctorIssue, DoctorIssueCode } from "./doctor-types.js";
 import { cleanNumberedGsdVariants } from "./repo-identity.js";
-import { milestonesDir, gsdRoot, resolveGsdRootFile, resolveMilestonePath } from "./paths.js";
+import { milestonesDir, gsdRoot, resolveGsdRootFile, resolveMilestonePath, resolveMilestoneFile } from "./paths.js";
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/doctor-runtime-checks.ts]: **Accepted** — queued for a code change.
**In progress** — applying the accepted fix.
**Verifying** — checking the applied changes.
**Resolved** — addressed in commit `d56e305`.

File: src/resources/extensions/gsd/doctor-runtime-checks.ts [hunk 5]
```diff
@@ -778,7 +778,17 @@ export async function checkRuntimeHealth(
...
+        // (legitimate in-flight planning). A `queued` row with no CONTEXT,
+        // ROADMAP, or SUMMARY is a phantom left by gsd_milestone_generate_id
+        // that was never planned — fall through so the missing-directory check
+        // below reports it as an orphan the user can clean up (#1524).
+        if (milestone.status === "queued") {
+          const hasContent = resolveMilestoneFile(basePath, milestone.id, "CONTEXT")
+            || resolveMilestoneFile(basePath, milestone.id, "ROADMAP")
+            || resolveMilestoneFile(basePath, milestone.id, "SUMMARY");
+          if (hasContent) continue;
+        }
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/doctor-runtime-checks.ts]: **Accepted** — queued for a code change.
**In progress** — applying the accepted fix.
**Verifying** — checking the applied changes.
**Resolved** — addressed in commit `d56e305`.

**Comment by jeremymcs** [comment]
Started an automated PR update.

Accepted review feedback is being applied.
Progress and final results will be posted in the related review thread(s).

File: src/resources/extensions/gsd/doctor-runtime-checks.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/doctor-runtime-checks.ts]: Addressed in commit `d56e305`.

The unused `resolveMilestoneFile` import is already removed — line 6 now imports only `resolveMilestonePath`, and no references to `resolveMilestoneFile` remain in the file (resolved in commit d56e305d).

File: src/resources/extensions/gsd/doctor-runtime-checks.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/doctor-runtime-checks.ts]: Addressed in commit `d56e305`.

The redundant `queued`/`hasContent` guard is already removed — the orphan DB-row check now runs the missing-directory test for all statuses via `!resolveMilestonePath(...)`, and the comment was corrected (commit d56e305d). Verified: doctor tests (d)/(e)/(f)/(g) covering this behavior all pass.

File: src/resources/extensions/gsd/doctor-runtime-checks.ts [hunk 6]
```diff
@@ -778,7 +778,12 @@ export async function checkRuntimeHealth(
...
     if (isDbAvailable()) {
       for (const milestone of getAllMilestones()) {
-        if (milestone.status === "queued") continue;
+        // Every milestone status, including `queued`, is subject to the
+        // missing-directory orphan check below. A legitimate in-flight queued
+        // milestone has a resolvable directory on disk, so resolveMilestonePath
+        // succeeds and it is not flagged. A `queued` phantom left by
+        // gsd_milestone_generate_id (no directory, no content) has no resolvable
+        // path and is correctly reported as an orphan to clean up (#1524).
         if (!resolveMilestonePath(basePath, milestone.id)) {
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/doctor-runtime-checks.ts]: **Accepted** — queued for a code change.
**In progress** — applying the accepted fix.
**Verifying** — checking the applied changes.
**Resolved** — addressed in commit `11fc160`.

File: src/resources/extensions/gsd/tests/doctor-orphan-milestone-4996.test.ts [hunk 7]
```diff
@@ -129,4 +129,36 @@ describe("gsd_doctor orphan milestone directory check (#4996)", () => {
...
+    // Legitimate in-flight planning: queued row with a CONTEXT file on disk.
+    insertMilestone({ id: "M005", status: "queued" });
+
+    const issues: DoctorIssue[] = [];
+    const fixes: string[] = [];
+    await checkRuntimeHealth(base, issues, fixes, () => false);
+
+    const orphan = issues.find(i => i.code === "orphan_milestone_db" && i.unitId === "M005");
+    assert.ok(!orphan, "queued row with content files is legitimate in-flight planning, not an orphan");
+  });
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/doctor-orphan-milestone-4996.test.ts:163]: **Accepted** — queued for a code change.
**In progress** — applying the accepted fix.
**Verifying** — checking the applied changes.
**Resolved** — addressed in commit `11fc160`.

**Comment by jeremymcs** [comment]
Started an automated PR update.

Accepted review feedback is being applied.
Progress and final results will be posted in the related review thread(s).

File: src/resources/extensions/gsd/doctor-runtime-checks.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/doctor-runtime-checks.ts]: Addressed in commit `305e685`.

The orphan DB-row check no longer uses `resolveMilestonePath()` (content-bearing) as the "directory exists" proxy. Commit 305e6855 adds `milestoneDirExists(basePath, milestoneId)` in `paths.ts` — a pure directory-**presence** check that returns true for a legacy `milestones/<MID>/` scaffold dir (e.g. an empty `slices/`) even when it holds no CONTEXT/ROADMAP. `doctor-runtime-checks.ts` now gates `orphan_milestone_db` on `!milestoneDirExists(...)`, so a legitimate in-flight queued milestone with only a scaffold directory is no longer flagged; `orphan_milestone_db` is reserved for the truly no-directory phantom case (#1524).

Verified by test `(h)` in `doctor-orphan-milestone-4996.test.ts` (queued row + empty on-disk dir → no orphan) and `(f)`/`(g)`, all passing.

File: src/resources/extensions/gsd/tests/doctor-orphan-milestone-4996.test.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/doctor-orphan-milestone-4996.test.ts:163]: Addressed in commit `305e685`.

Added regression test `(h)` in `doctor-orphan-milestone-4996.test.ts` covering exactly this case: a `queued` DB row whose milestone directory exists on disk as a scaffold-only dir (`stubDir` creates `.gsd/milestones/M006/slices` with no content), asserting it is **not** reported as `orphan_milestone_db`. This guards against the doctor warning during normal in-flight planning when the workflow prompts create the milestone directory before any CONTEXT/ROADMAP is written. Test passes alongside the existing content-file case `(g)` and the phantom (no-dir) case `(f)`.

File: src/resources/extensions/gsd/state/derive/from-db.ts [hunk 8]
```diff
@@ -129,6 +131,26 @@ function milestoneArtifactExistsInResolvedDir(
...
+// PROJECT.md "Milestone Sequence". A content-less queued milestone that appears
+// here is a real, not-yet-planned roadmap stage (e.g. the first milestone right
+// after deep-project setup) and is safe to promote to active. A content-less
+// queued row that is NOT listed here is a phantom left by
+// gsd_milestone_generate_id that was never made part of the roadmap (#1524) and
+// must not be promoted. Returns an empty set when PROJECT.md is absent or
+// unparsable, which keeps phantom-only repos out of the promotion path.
+function loadProjectSequenceIds(basePath: string): Set<string> {
+  const projectPath = resolveGsdRootFile(basePath, 'PROJECT');
+  if (!existsSync(projectPath)) return new Set();
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: **Accepted** — queued for a code change.
**In progress** — applying the accepted fix.
**Verifying** — checking the applied changes.
**Resolved** — addressed in commit `11fc160`.

File: src/resources/extensions/gsd/state/derive/from-db.ts [hunk 9]
```diff
@@ -129,6 +131,26 @@ function milestoneArtifactExistsInResolvedDir(
...
+  try {
+    const parsed = parseProject(readFileSync(projectPath, 'utf-8'));
+    return new Set(parsed.milestones.map((m) => m.id));
+  } catch (e) {
+    logWarning('state', `failed to parse PROJECT.md milestone sequence: ${(e as Error).message}`);
+    return new Set();
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: **Accepted** — queued for a code change.
**In progress** — applying the accepted fix.
**Verifying** — checking the applied changes.
**Resolved** — addressed in commit `11fc160`.

**Comment by jeremymcs** [comment]
Started an automated PR update.

Accepted review feedback is being applied.
Progress and final results will be posted in the related review thread(s).

File: src/resources/extensions/gsd/doctor-runtime-checks.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/doctor-runtime-checks.ts]: Addressed in commit `11fc160`.

File: src/resources/extensions/gsd/tests/doctor-orphan-milestone-4996.test.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/doctor-orphan-milestone-4996.test.ts:163]: Addressed in commit `11fc160`.

File: src/resources/extensions/gsd/state/derive/from-db.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: Addressed in commit `11fc160`.

Line 144 already returns `new Set<string>()` (explicitly typed) rather than `new Set()`, resolving the `Set<any>` type-safety concern. No further change was needed; verified with `tsc --noEmit`.

File: src/resources/extensions/gsd/state/derive/from-db.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/state/derive/from-db.ts]: Addressed in commit `11fc160`.

Line 150 already returns `new Set<string>()` (explicitly typed) in the catch block, preserving the `Set<string>` contract. No further change was needed; verified with `tsc --noEmit`.
